addons/test_mail_full/tests/test_mail_thread_phone.py

```diff
@@ -65,3 +65,18 @@ def test_search_phone_mobile_search(self):
         results = self.env['mail.test.sms.bl'].search([('phone_mobile_search', 'not ilike', '0475')])
         results = self.env['mail.test.sms.bl'].search([('phone_mobile_search', 'not ilike', '101')])
         results = self.env['mail.test.sms.bl'].search([('phone_mobile_search', 'not ilike', '606')])
+
+    @users('employee')
+    def test_search_phone_mobile_search_boolean(self):
+        test_phone_records = self.test_phone_records.with_env(self.env)
+
+        # test Falsy -> is set / is not set
+        for test_values in [False, '', ' ']:
+            results = self.env['mail.test.sms.bl'].search([('phone_mobile_search', '=', test_values)])
+            self.assertFalse(results)
+            results = self.env['mail.test.sms.bl'].search([('phone_mobile_search', '!=', test_values)])
+            self.assertEqual(results, test_phone_records)
+
+        # is not set -> != True
+        results = self.env['mail.test.sms.bl'].search([('phone_mobile_search', '!=', True)])
+        self.assertFalse(results)
```

Comment:
with my proposition above, the "opposite" will work `(..., '=', True)` (currently it return same result as `... == False`)

could be nice to add a test